frontend/www/js/omegaup/problem/edit.js

```diff
@@ -330,6 +331,109 @@ OmegaUp.on('ready', function() {
+            if (solutions.hasOwnProperty(language)) {
+              solutionEdit.updateAndRefresh(solutions[language]);
+              return;
+            }
+            API.Problem.solution({
+                         'problem_alias': problemAlias,
+                         'lang': language,
+                       })
+                .then(function(response) {
+                  if (!response.exists || !response.solution) {
+                    return;
+                  }
+                  if (response.solution.language !== language) {
+                    response.solution.markdown = '';
+                  }
+                  solutionEdit.solutions[language] = response.solution.markdown;
+                  solutionEdit.updateAndRefresh(response.solution.markdown);
+                })
+                .fail(UI.apiError);
+          },
+          'edit-solution': function(solutions, commitMessage, currentLanguage) {
+            let promises = [];
+            for (const lang in solutions) {
+              if (!solutions.hasOwnProperty(lang)) continue;
+              if (solutions[lang] === solutionEdit.solutions[lang]) continue;
+              promises.push(new Promise(function(resolve, reject) {
+                API.Problem.updateSolution({
+                             problem_alias: problemAlias,
+                             solution: solutions[lang],
+                             message: commitMessage,
+                             lang: lang,
+                           })
+                    .then(function(response) { resolve(response); })
+                    .fail(UI.apiError);
+              }));
+            }
+            Promise.all(promises)
+                .then(function() {
+                  UI.success(T.problemEditUpdatedSuccessfully);
+                })
+                .catch(function(error) { UI.apiError(error); });
+          },
+        },
+      });
+    },
+    mounted: function() {
+      const markdownConverter =
+          UI.markdownConverter({preview: true, imageMapping: {}});
+      this.markdownEditor = new Markdown.Editor(markdownConverter, '-solution');
+      this.markdownEditor.run();
+    },
+    data: {
+      markdownContents: null,
+      markdownPreview: '',
+      markdownEditor: null,
+      initialLanguage: null,
+      solutions: {},
+    },
+    methods: {
+      updateAndRefresh(markdown) {
+        this.markdownContents = markdown;
+        this.markdownPreview =
+            this.markdownEditor.getConverter().makeHtml(markdown);
+      },
+      getInitialContents() {
+        let self = this;
+        API.Problem.solution({
+                     'problem_alias': problemAlias,
+                   })
+            .then(function(response) {
+              if (!response.exists || !response.solution) {
+                return;
+              }
+              const lang = response.solution.language;
+              self.initialLanguage = lang;
```

Comment:
qué pasa si en vez de cambiar los `props`, cambias los miembros directamente?

```suggestion
              self.solutionEdit.currentLanguage = lang;
```

igual con los demás props excepto tal vez `solutions` porque ese necesitas tener una copia aparte para saber si hay necesidad de actualizarlo. Eso te permitiría eliminar un par de watchers.

Reply:
`currentLanguage` es una variable en `data` dentro de `StatementEdit`, si hago que sea un `prop`, no podrá ser modificado en el componente, [`Vue` se queja de eso](https://stackoverflow.com/questions/40780730/vue-js-changing-props). Ya lo había intentado.

Reply:
argh, no vi que dentro del componente también modificas `currentLanguage`. Olvida todo esto.